packages/fetchai/skills/erc1155_client/handlers.py

```diff
@@ -0,0 +1,237 @@
+
+from typing import List, Optional, cast
+
+from aea.configurations.base import ProtocolId
+from aea.decision_maker.messages.transaction import TransactionMessage
+from aea.protocols.base import Message
+from aea.skills.base import Handler
+
+from packages.fetchai.protocols.fipa.message import FIPAMessage
+from packages.fetchai.protocols.fipa.serialization import FIPASerializer
+from packages.fetchai.protocols.oef.message import OEFMessage
+from packages.fetchai.skills.erc1155_client.strategy import Strategy
+
+
+class FIPAHandler(Handler):
+    """This class implements a FIPA handler."""
+
+    SUPPORTED_PROTOCOL = FIPAMessage.protocol_id  # type: Optional[ProtocolId]
+
+    def setup(self) -> None:
+        """
+        Implement the setup.
+
+        :return: None
+        """
+        pass
+
+    def handle(self, message: Message) -> None:
+        """
+        Implement the reaction to a message.
+
+        :param message: the message
+        :return: None
+        """
+        # convenience representations
+        fipa_msg = cast(FIPAMessage, message)
+
+        # handle message
+        if fipa_msg.performative == FIPAMessage.Performative.INFORM:
+            self._handle_inform(fipa_msg)
+
+    def teardown(self) -> None:
+        """
+        Implement the handler teardown.
+
+        :return: None
+        """
+        pass
+
+    def _handle_inform(self, msg: FIPAMessage) -> None:
+        """
+        Handle the match inform.
+
+        :param msg: the message
+        :return: None
+        """
+        self.context.logger.info(
+            "[{}]: received INFORM from sender={}".format(
+                self.context.agent_name, msg.counterparty[-5:]
+            )
+        )
+        if len(msg.info.keys()) >= 1:
+            data = msg.info
+            if "contract" in data.keys():
+                contract = self.context.contracts.erc1155
+                contract.set_instance(
+                    ledger_api=self.context.ledger_apis.apis.get("ethereum")
+                )
+                contract.set_address(
+                    ledger_api=self.context.ledger_apis.apis.get("ethereum"),
+                    contract_address=data["contract"],
+                )
+                contract.item_ids = data["item_ids"]
+                self.context.shared_state["counterparty"] = msg.counterparty
+                tx = contract.get_hash_single_transaction(
+                    from_address=msg.counterparty,
+                    to_address=self.context.agent_address,
+                    item_id=contract.item_ids[0],
+                    from_supply=10,
+                    to_supply=0,
+                    value=0,
+                    trade_nonce=500,
+                    ledger_api=self.context.ledger_apis.apis.get("ethereum"),
+                    skill_callback_id=self.context.skill_id,
+                )
+
+                self.context.decision_maker_message_queue.put_nowait(tx)
```

Comment:
We request from the contract to hash a transaction and then we pass this to the decision maker for signing.

Reply:
could add a logger statement here with a similar message so its in code